cfgov/agreements/management/commands/util.py

```diff
@@ -1,64 +1,71 @@
 import os
 
-from agreements.models import Agreement, Issuer
+from agreements.models import Issuer
+import boto3
+
 from slugify import slugify
 
 
-def clear_tables():
-    Agreement.objects.all().delete()
+def s3_safe_key(path, prefix=''):
+    key = prefix + path.encode('utf-8')
+    key = key.replace(' ', '_')
+    key = key.replace('%', '')
+    key = key.replace(';', '')
+    return key
+
+
+def upload_to_s3(pdf_obj, s3_key):
+    AWS_STORAGE_BUCKET_NAME = os.environ.get('AWS_STORAGE_BUCKET_NAME')
+    AWS_S3_ACCESS_KEY_ID = os.environ.get('AWS_S3_ACCESS_KEY_ID')
+    AWS_S3_SECRET_ACCESS_KEY = os.environ.get('AWS_S3_SECRET_ACCESS_KEY')
 
+    s3_client = boto3.client('s3',
+                             aws_access_key_id=AWS_S3_ACCESS_KEY_ID,
+                             aws_secret_access_key=AWS_S3_SECRET_ACCESS_KEY)
 
-def get_file_size(path):
-    return os.path.getsize(path)
+    s3_client.upload_fileobj(pdf_obj, AWS_STORAGE_BUCKET_NAME, s3_key)
 
 
-def update_issuer(name):
+def get_issuer(name):
+    slug = slugify(name, to_lower=True)
     try:
-        issuer = Issuer.objects.get(name=name)
+        issuer = Issuer.objects.get(slug=slug)
     except Issuer.DoesNotExist:
-        issuer = Issuer()
-    except Issuer.MultipleObjectsReturned:
-        raise Exception('Multiple Issuers exist for %s' % name)
+        issuer = Issuer(slug=slug, name=name)
+        issuer.save()
+    return issuer
 
-    issuer.name = name
-    issuer.slug = slugify(name, to_lower=True)
-    issuer.save()
 
-    return issuer
+def save_agreement(agreements_zip, raw_path, filename_encoding,
+                   outfile, upload=False):
+    uri_hostname = 'http://files.consumerfinance.gov'
+    s3_prefix = '/a/assets/credit-card-agreements/pdf/'
```

Comment:
the `s3_prefix` var shouldn't have a leading slash. S3 uploads wouldn't work for me unless I removed it.

This also requires an adjustment to the `save_agreement` function, to keep the uri in sync; something like

```
    agreement = issuer.agreement_set.create(
        file_name=filename,
        size=int(zipinfo.file_size),
        uri="{}/{}".format(uri_hostname, s3_key),
        description=filename)
```

Reply:
oh, interesting-- I wonder why that didn't bite me

Reply:
I was testing from laptop, not a server?

Reply:
Latest commit addresses this